src/com/ConnectionInfoPublisher.cpp

```diff
@@ -0,0 +1,93 @@
+#include "ConnectionInfoPublisher.hpp"
+#include <chrono>
+#include <thread>
+#include <boost/uuid/name_generator.hpp>
+#include <boost/uuid/string_generator.hpp>
+#include <boost/uuid/uuid_io.hpp>
+#include <boost/filesystem.hpp>
+
+ConnectionInfoPublisher::ConnectionInfoPublisher(std::string acceptorName,
+                                                 std::string requesterName,
+                                                 int rank,
+                                                 std::string addressDirectory)
+  :
+  acceptorName(acceptorName),
```

Comment:
You should `std::move` the strings straight into the members in this an the following constructor.

```cpp
acceptorName(std::move(acceptorName)),
```

Also consider delegating the strings, part to the following constructor.